homeassistant/components/statistics/sensor.py

```diff
@@ -460,11 +460,25 @@ def _add_state_to_queue(self, new_state: State) -> None:
             )
             return
 
-        self._unit_of_measurement = self._derive_unit_of_measurement(new_state)
+        self._calculate_state_attributes(new_state)
 
-    def _derive_unit_of_measurement(self, new_state: State) -> str | None:
+    def _calculate_state_attributes(self, new_state: State) -> None:
```

Comment:
In light of this new function I suggest to rename the existing function `_update_attributes(self)`. Those names clash currently. How about something like `_update_additional_attributes(self)` or `_calculate_custom_attributes(self)`

Reply:
We should update both `_update_attributes` and the `attributes` `dict` to make it clear it's about extra state attributes.
Maybe that can happen in a separate PR, to not pollute this PR?

Reply:
I agree, let's "clean" this in a follow-up PR.
I didn't even consider `self.attributes` nor the `_update_attributes` method

Reply:
Feels like an important change for clarity, but good on the extra PR

Reply:
https://github.com/home-assistant/core/pull/129353